extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkus.rest.client.reactive.beanparam;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.URI;
+
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.test.common.http.TestHTTPResource;
+
+public class BeanPathParamTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest();
+
+    @TestHTTPResource
+    URI baseUri;
+
+    @Test
+    void shouldPassPathParamFromBeanParam() {
+        Client client = RestClientBuilder.newBuilder().baseUri(baseUri).build(Client.class);
+        assertThat(client.getWithBeanParam(new MyBeanParam("123"))).isEqualTo("it works!");
+
+    }
+
+    @Path("/my/{id}/resource")
+    public interface Client {
+        @GET
+        String getWithBeanParam(@BeanParam MyBeanParam beanParam);
+    }
```

Comment:
Should we not also add more methods that have `@PathParam` themselves?

Reply:
👍🏻 it's worth checking if they mix well together, will do

Reply:
done